app/src/main/java/tech/gruppone/stalker/app/utility/CurrentSessionSingleton.java

```diff
@@ -68,53 +73,60 @@ public void setJwt(@NonNull String token) {
   }
 
   public void setOrganizationList(@NonNull List<Organization> orgList) {
-    organizations.postValue(orgList);
+    @SuppressLint("UseSparseArrays")
+    Map<Integer, LiveData<Organization>> map = new TreeMap<>();
+
+    for (Organization organization : orgList) {
+      map.put(organization.getId(), new MutableLiveData<>(organization));
+    }
+
+    organizations.postValue(map);
   }
 
   @NonNull
-  public LiveData<List<Organization>> getOrganizations() {
+  public LiveData<Map<Integer, LiveData<Organization>>> getOrganizations() {
     return organizations;
   }
 
   @NonNull
-  public Organization getOrganization(int organizationId) throws OrganizationNotFoundException {
-    for (Organization organization : Objects.requireNonNull(getOrganizations().getValue())) {
-      if (organization.getId() == organizationId) {
-        return organization;
-      }
+  public LiveData<Organization> getOrganization(int organizationId)
+      throws OrganizationNotFoundException {
+    LiveData<Organization> organization =
+        requireNonNull(organizations.getValue()).get(organizationId);
+
+    if (organization == null) {
+      throw new OrganizationNotFoundException(
+          "There isn't an organization with the given id of " + organizationId);
     }
 
-    throw new OrganizationNotFoundException("Missing organization with id: " + organizationId);
+    return organization;
   }
 
   public boolean zeroOrganizations() {
-    return organizations.getValue() == null || organizations.getValue().isEmpty();
+    return requireNonNull(organizations.getValue()).isEmpty();
   }
 
-  public void connectOrganization(int organizationId) {
-    for (Organization organization : Objects.requireNonNull(organizations.getValue())) {
-      if (organization.getId() == organizationId) {
-        organization.setConnected(true);
+  public void connectOrganization(int organizationId) throws OrganizationNotFoundException {
+    // Necessary cast, since java generics are invariant
+    // We wouldn't need it if they could be covariant (in kotlin, for example)
```

Comment:
"code tells you how, comments tell you why [this programming language sucks]" cit. Riccardo Agatea